### PR TITLE
Force babel to transpile ts and tsx files

### LINKED ISSUE #50: Is distributing as ES6 intentional
Hi!

This library is currently distributed as ES6 which is a bit troublesome as we need to support IE11 and would like to avoid transpiling `node_modules` if possible. We did not experience this with `react-stripe-elements`, so this was an unexpected migration step.

Would you consider distributing the package as ES5 instead?

PS. Thank you for an excellent developer experience using Stripe.

### PR DESCRIPTION
**Pull request by dweedon-stripe**
### Summary & motivation
Ensure that babel transpiles TypeScript files to ES5. 

Fixes #50

### Testing & documentation

Investigated build output and tested locally.

### CODE REVIEW

File: rollup.config.js [hunk 1]
```diff
@@ -14,7 +14,14 @@ export default [
       {file: pkg.main, format: 'cjs'},
       {file: pkg.module, format: 'es'},
     ],
-    plugins: [ts(), resolve(), babel(), commonjs()],
+    plugins: [
+      ts(),
+      resolve(),
+      babel({
+        extensions: ['.ts', '.tsx'],
```
Comment (christopher-stripe) [inline rollup.config.js]: (2) We `allowJS` in our project, so it might be safer to achieve this with some sort of ignore pattern for `node_modules`.

Comment (dweedon-stripe) [reply rollup.config.js]: Babel will still ignore `node_modules` by default, but I can extend `extensions` to include `js` and `jsx`.

Comment (hofman-stripe) [reply rollup.config.js]: I have mixed feeling regarding transpiling down the module version. IMO, publishing only ES5 code is the source of a lot of inefficiencies on the web today.

Comment (christopher-stripe) [reply rollup.config.js]: > Babel will still ignore node_modules by default, but I can extend extensions to include js and jsx.

Oh ok great.

> I have mixed feeling regarding transpiling down the module version. IMO, publishing only ES5 code is the source of a lot of inefficiencies on the web today.

Yeah this is a bummer, but I don't think we have much of a choice (don't want users to need to compile node_modules).

**Pull request merged**